pkg/scheduler/scheduler.go

```diff
@@ -149,14 +190,19 @@ func (s *Scheduler) schedule(ctx context.Context) {
 			//      evaluated in the same cycle is not taken into account.
 			//   2. An already admitted workload from a different cluster queue is borrowing, since all workloads
 			//      evaluated in the current cycle will compete for the resources that are not borrowed.
-			if usedCohorts.Has(cq.Cohort.Name) && (e.assignment.Borrows() || cq.Cohort.HasBorrowingQueues()) {
-				e.status = skipped
-				e.inadmissibleMsg = "other workloads in the cohort were prioritized"
-				continue
+			if cycleCohortUsage.HasConflictWith(cq.Cohort.Name, e.assignment.Usage) && (e.assignment.Borrows() || cq.Cohort.HasBorrowingQueues()) {
+				// checkIfStllFits
+				sum := cycleCohortUsage.GetSumWith(cq.Cohort.Name, e.assignment.Usage)
+				if !cq.Cohort.CanFit(sum) {
```

Comment:
I'm wondering if this check is needed because, as I can see, the flavorAssigner already checks:

https://github.com/kubernetes-sigs/kueue/blob/3e3ebb65e61845fc519e4b7a7bea9fc1cdf4c767/pkg/scheduler/flavorassigner/flavorassigner.go#L470-L484

If this check is needed, can you clarify?

Reply:
`"no overadmission while borrowing"` is covering it

Reply:
SGTM